### PR TITLE
add previous and next button to History and Logbook

### PR DESCRIPTION
**Pull request by LiBe-net**
## Proposed change

add previous and next buttons to History and Logbook:
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-date-range-picker.ts [hunk 1]
```diff
@@ -317,6 +331,28 @@ export class HaDateRangePicker extends LitElement {
...
+    const dateRange = [
+      new Date(this.startDate.getTime() - diff),
+      new Date(this.startDate.getTime() - 1),
+    ];
+    const dateRangePicker = this._dateRangePicker;
+    dateRangePicker.clickRange(dateRange);
+    dateRangePicker.clickedApply();
+  }
+
   private _setDateRange(ev: CustomEvent<ActionDetail>) {
```
Comment (wendevlin) [inline src/components/ha-date-range-picker.ts:356]: Please use `date-fns` to calculate the time.

I understand your +/- 1, technically it is correct but I wouldn't go into that detail:
- 2pm - 5pm -> 11am - 2pm
Is way easier to read for the user

File: src/components/ha-date-range-picker.ts [hunk 2]
```diff
@@ -249,6 +251,12 @@ export class HaDateRangePicker extends LitElement {
         <div slot="input" class="date-range-inputs" @click=${this._handleClick}>
           ${!this.minimal
             ? html`<ha-svg-icon .path=${mdiCalendar}></ha-svg-icon>
+                <ha-icon-button-prev
+                  .label=${this.hass.localize("ui.common.previous")}
+                  class="prev"
+                  @click=${this._handlePrev}
+                >
+                </ha-icon-button-prev>
```
Comment (wendevlin) [inline src/components/ha-date-range-picker.ts:259]: Please hide prev and next buttons on mobile, it takes too much space, look how it is done with `ha-svg-icon`

**Review by wendevlin — CHANGES_REQUESTED** [review]
Thank you, really nice improvement.

I just made some comments.

Please [set](https://docs.github.com/en/get-started/getting-started-with-git/setting-your-username-in-git) your git config name. Currently it is set to `Your Name`

File: src/components/ha-date-range-picker.ts (hunk 1 above)
Comment (boern99) [reply src/components/ha-date-range-picker.ts:356]: I will change the date-conversion using date-fns.
Yes, the reason for +1 and -1 is that the time range is diplayed 0:00 to 23:59 for today, this month, week etc. The End Date is missing 1 millisecond.
So, if I use the real timespan to shift Previous multiple times it does not move exact days.

File: src/components/ha-date-range-picker.ts (hunk 2 above)
Comment (boern99) [reply src/components/ha-date-range-picker.ts:259]: Maybe a stupid question, why is there a second Date-Entry for EndDate? Both: StartDate and Enddate load the exact same Daterange-Picker which includes Start and Enddate? Thought about to remove the EndDate for mobile. 
If i disable the prev and next buttons they can not be used on small displays. Maybe I found another solution: changing the media-query for the mentioned ha-svg-icon {display: none} solved it.

File: src/components/ha-date-range-picker.ts [hunk 3]
```diff
@@ -413,7 +456,7 @@ export class HaDateRangePicker extends LitElement {
         }
       }
 
-      @media only screen and (max-width: 500px) {
+      @media only screen and (max-width: 560px) {
```
Comment (boern99) [inline src/components/ha-date-range-picker.ts:459]: Only had troubles on Resolutions between 500 and 560px.
I added 560 here, as min-width:inherit shrinks the date to fit ..

File: src/components/ha-date-range-picker.ts (hunk 2 above)
Comment (wendevlin) [reply src/components/ha-date-range-picker.ts:262]: It is there to see the time range, but sure we could show the time range in a more compact way then an input field. If you want to try it out, please open a seperate PR.
For now please hide prev and next buttons on mobile and in the next steps we can improve it, maybe with help of UX.

**Review by wendevlin — CHANGES_REQUESTED** [review]

File: src/components/ha-date-range-picker.ts (hunk 3 above)
Comment (boern99) [reply src/components/ha-date-range-picker.ts:459]: @wendevlin: instead hiding it on mobile: this change shrinks the date to fit. If I missed something important here: for sure hiding would be also possible in the first step if really necessary ...

Comment (wendevlin) [reply src/components/ha-date-range-picker.ts:459]: The arrows still take too much space on small phones, so I would like to hide it for now and improve the whole input mask with another PR.

File: src/components/ha-date-range-picker.ts (hunk 2 above)
Comment (boern99) [reply src/components/ha-date-range-picker.ts:263]: ok, thank you

**Review by wendevlin — APPROVED** [review]

**Pull request merged**